src/stores/ClientPresenceStore.js

```diff
@@ -0,0 +1,55 @@
+const PresenceStore = require('./DataStore');
+const Collection = require('../util/Collection');
+const Constants = require('../util/Constants');
+const { Presence } = require('../structures/Presence');
+
+class ClientPresenceStore extends PresenceStore {
+  constructor(...args) {
+    super(...args);
+    this.clientPresence = new Presence({
+      status: 'online',
+      afk: false,
+      since: null,
+      activity: null,
+    });
+  }
+
+  async setClientPresence({ status, since, afk, activity }) {
+    const applicationID = activity.application ? activity.application.id || activity.application : null;
+    let assets = new Collection();
+    if (activity.assets && applicationID) {
+      try {
+        const a = await this.client.api.applications(applicationID).assets.get();
+        for (const asset of a) assets.set(asset.name, asset.id);
+      } catch (err) {} // eslint-disable-line no-empty
+    }
+
+    const packet = {
+      afk: afk != null ? afk : false, // eslint-disable-line eqeqeq
+      since: since != null ? since : null, // eslint-disable-line eqeqeq
+      status: status || this.clientPresence.status,
+      game: activity ? {
+        type: Constants.ActivityTypes.indexOf(activity.type || 'PLAYING') || activity.type || 0,
```

Comment:
You can no longer pass a numeric type here (except 0) as the indexOf returns `-1` in that case which is not falsy.